### PR TITLE
prepare the changelog for the 6.0.0 release

### PR DESCRIPTION
**Pull request by tgummerer**
This includes a breaking change from https://github.com/pulumi/actions/pull/1118, so we should release a new major version.

### CODE REVIEW

**Review by justinvp — CHANGES_REQUESTED** [review]
Before we release this, should we address Komal's comment at https://github.com/pulumi/actions/pull/1118#issuecomment-2021328817?

> Looks like we at least have the refresh option for [up](https://github.com/pulumi/pulumi/commit/f32b16ac0e510b7bbb21bbb41ff906f7f140b9e4) and [preview](https://github.com/pulumi/pulumi/issues/12740) in nodejs automation api, but would be great to get this supported for `destroy` as well to have consistency across commands.

Otherwise, it looks like if you're running `destroy` with the `refresh` option set, we no longer do a refresh at all.

**Comment by tgummerer** [comment]
https://github.com/pulumi/pulumi/pull/17209 has the changes.  We can merge this after merging and releasing that.

**Comment by tgummerer** [comment]
Indeed, thank you! I submitted a PR to update the pulumi version in this repo here: https://github.com/pulumi/actions/pull/1287

Once that's merged I think we can go ahead with the release!

**Review by julienp — APPROVED** [review]

File: CHANGELOG.md
Comment (fitz-vivodyne) [inline CHANGELOG.md:9]:
```suggestion
- feat: (breaking change) Update the `refresh` option to run commands with the
```

Comment (tgummerer) [reply CHANGELOG.md]: Thanks!

File: CHANGELOG.md
Comment (justinvp) [inline CHANGELOG.md:7]: Needs updated date.

**Review by justinvp — APPROVED** [review]
LGTM otherwise

File: CHANGELOG.md
Comment (tgummerer) [reply CHANGELOG.md]: Done, thanks!

**Pull request merged**
